app/services/media_storage/azure_adapter.rb

```diff
@@ -0,0 +1,92 @@
+module MediaStorage
+  class AzureAdapter < AbstractAdapter
+    DEFAULT_EXPIRES_IN = 3 # time in minutes
+
+    def initialize(opts={})
+      @storage_account_name = opts[:azure_storage_account]
+      @container = opts[:azure_storage_container] || Rails.env
+      @get_expiration = opts.dig(:expiration, :get) || DEFAULT_EXPIRES_IN
+      @put_expiration = opts.dig(:expiration, :put) || DEFAULT_EXPIRES_IN
+
+      @client = Azure::Storage::Blob::BlobService.create(
+        storage_account_name: opts[:azure_storage_account],
+        storage_access_key: opts[:azure_storage_access_key]
+      )
+      @signer = Azure::Storage::Common::Core::Auth::SharedAccessSignature.new(
+        opts[:azure_storage_account],
+        opts[:azure_storage_access_key]
+      )
+    end
+
+    def stored_path(content_type, medium_type, *path_prefix)
+      extension = get_extension(content_type)
+      path = "#{medium_type}/"
```

Comment:
Ah yes. I think I initially misunderstood the desired behavior of this method, but Im pretty sure I see it now.

Reply:
Method and specs updated 👍 
I've added the env var `AZURE_PREFIX` because the azure adapter `stored_path` method will use the container name to generate the part of the URL that specifies the environment - 
https://github.com/zooniverse/panoptes/blob/9f6a1429464678d836ff1c3e1ac44a166d53418d/app/services/media_storage/azure_adapter.rb#L27
which means that what we pass in as a prefix will be different from the AWS prefix.

In addition, we might end up wanting the azure prefix to be different from the aws prefix during testing / while we have both adapters running simultaneously.